mmengine/dist/dist.py

```diff
@@ -307,7 +307,8 @@ def broadcast(data: Tensor,
         input_device = get_data_device(data)
         backend_device = get_comm_device(group)
         data_on_device = cast_data_device(data, backend_device)
-
+        if isinstance(data_on_device, torch.Tensor):
```

Comment:
Is this check necessary? It seems `broadcast` only accepts tensor type inputs

Reply:
This check is added to pass `mypy` check. Is there any other better solutions?

Reply:
Will `data_on_device: Tensor = cast_data_device(data, backend_device)` or `assert isinstance(data_on_device, torch.Tensor)` work? If not, I think we should use `# type: ignore` or `if xxx; else raise TypeError` to avoid this confuse `if-else`

Reply:
This check fixes lint because it passes some other types like Map or Iter. If you have better ways to avoid confusion then please replace this.